### PR TITLE
feat: combo box multi select

### PR DESCRIPTION
**Pull request by ninariccimarie**

### CODE REVIEW

File: src/@next/ComboBox/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  const inputRef = useRef(null);
+  const textInputContext = useComboboxTextInput();
+  const { onFocus, setTextInputWidth } = textInputContext;
+
+  useEffect(() => {
+    setTextInputWidth(inputRef.current.getBoundingClientRect().width);
+  }, [inputRef, setTextInputWidth]);
+
+  const handleChange = (value: string) => {
+    if (onChange) onChange(value);
```
Comment (marthayaputra) [inline src/@next/ComboBox/components/TextInput/TextInput.tsx]: can we use curly bracket or optional chain here?

File: src/@next/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  const optionContextValue = {
+    allowMultiple,
+  };
+
+  return (
+    <Popover
+      active={popoverActive}
+      activator={
+        <ComboboxTextInputContext.Provider value={textInputContextValue}>
```
Comment (marthayaputra) [inline src/@next/ComboBox/ComboBox.tsx]: Any reason we use Context here? Also for other contexts

File: src/@next/ComboBox/components/OptionList/OptionList.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  isEmpty,
+  noOptionsMessage,
+  onSelect,
+}: OptionListProps) => {
+  const textInputContext = useComboboxTextInput();
+  const { textInputWidth } = textInputContext;
+  const hasChildren = !isEmpty;
+
+  const onOptionSelect = (value: string) => {
+    if (onSelect) onSelect(value);
```
Comment (marthayaputra) [inline src/@next/ComboBox/components/OptionList/OptionList.tsx]: can we have curly bracket or optional chaining, I think we should set ESLint rule for this

**Review by marthayaputra — CHANGES_REQUESTED** [review]

File: src/@next/ComboBox/ComboBox.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+import { Combobox, ComboboxProps } from './Combobox';
+import { StyledTag } from './comboboxStoryHelper/TagStyle';
+
+(Combobox as React.FunctionComponent<ComboboxProps>).displayName = 'Combobox';
+
+export default {
+  title: '@next/Combobox',
+  component: Combobox,
+  decorators: [
```
Comment (marthayaputra) [inline src/@next/ComboBox/ComboBox.stories.tsx]: can use `withGlintsPortalContainer` decorator here

File: src/@next/Combobox/components/OptionList/OptionListStyle.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    margin-bottom: ${space4};
+
+    &:hover {
+      background: ${Neutral.B99};
+      border-radius: ${borderRadius4};
+      cursor: pointer;
+    }
+
+    &[data-active='true'] {
+      background: ${Blue.S08};
```
Comment (chiahou) [inline src/@next/Combobox/components/OptionList/OptionListStyle.ts]: Design change: bg color remains unchange in the active state and thus should be removed.

File: src/@next/Combobox/components/OptionList/OptionListStyle.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import styled from 'styled-components';
+import { borderRadius4 } from '../../../utilities/borderRadius';
+import { Blue, Neutral } from '../../../utilities/colors';
+import { space4, space8 } from '../../../utilities/spacing';
+
+interface OptionListContainerProps {
+  textInputWidth?: number;
+}
+
+export const OptionListContainer = styled.div<OptionListContainerProps>`
```
Comment (chiahou) [inline src/@next/Combobox/components/OptionList/OptionListStyle.ts:10]: the position of OptionListContainer doesn't align with TextInput when the screen width is smaller than the component's width. Could you take a look at this issue?

Comment (ninariccimarie) [reply src/@next/Combobox/components/OptionList/OptionListStyle.ts:10]: @chiahou it had to do with how we defined the width for TextInput. thanks for noticing. I've addressed above. You'll need to refresh when you try to change your screen size to make sure both have same width

**Review by chiahou — APPROVED** [review]
qc passed, thanks!

**Review by marthayaputra — APPROVED** [review]

**Pull request merged**